src/coreclr/vm/stringliteralmap.cpp

```diff
@@ -456,7 +456,7 @@ STRINGREF AllocateStringObject(EEStringData *pStringData)
     // Create the COM+ string object.
     DWORD cCount = pStringData->GetCharCount();
 
-    STRINGREF strObj = AllocateString(cCount);
+    STRINGREF strObj = AllocateString(cCount, pinned ? GC_ALLOC_PINNED_OBJECT_HEAP : GC_ALLOC_NO_FLAGS);
```

Comment:
Maybe use `GC_ALLOC_ZEROING_OPTIONAL` as well? - since you are going to write over the content anyways.
At least when this goes into POH. POH does not use allocation contexts, so there will be zeroing.

Reply:
Yeah we decided to give up on this approach with POH, I am trying to make a prototype with `VirtualAlloc`

Reply:
Yes, I noticed later in the PR that you've decided to build a separate allocator for immortal strings. I think POH might have worked too and be simpler.
Anyways, in case you rethink after trying alternatives or use POH for unloadable literals, - makes sense to disable zeroing.

CC @MichalStrehovsky